src/DataAPI.jl

```diff
@@ -287,4 +287,11 @@ using a `sink` function to materialize the table.
 """
 function allcombinations end
 
+"""
+    getmetadata(x, ...)
+
+Return metadata associated with object `x` or `nothing` if it doest not support metadata.
```

Comment:
Let's give more guarantees?
```suggestion
Return metadata associated with object `x` as an `AbstractDict{String,Any}` object
(or an object implementing the same interface), or `nothing` if `x` does not support metadata.
```

I think requiring `AbstractDict{String,String}` would be too strict as it can be useful to store e.g. the observation date, the duration of observation, a numeric index, the parent population size...

Reply:
Yes - it would be useful. We can make this more flexible. The only thing is that Arrow.jl now only accepts strings, so the question is if this can be lifted there? The point is that I wanted, in particular, to ensure that if in DataFrames.jl we set some metadata it can be stored and retrieved by Arrow.jl.

Reply:
There's a tension here. I'd favor allowing any element type as it would be too bad to limit what we can do in Julia just because Arrow doesn't support it. This would be similar to allowing only element types supported by Arrow in data frame columns. :-) Maybe Arrow.jl can call `string` on each element before saving it?

Reply:
The Arrow.jl `{String, String}` restriction is strict, but I agree that we could just call `string` on the values to convert (I think we already do this in fact). I think we could maybe mention that some systems may assume `Dict{String, String}`, and so it's suggested but not a hard requirement? Something like that?

Reply:
OK - I have updated the PR following the comments.
@nalimilan - could you please, in particular, check wording of the docstring. Thank you!